src/lighteval/models/endpoints/endpoint_model.py

```diff
@@ -574,10 +582,18 @@ def greedy_until(
 
         return dataset.get_original_order(results)
 
+    @cached("predictions")
     def loglikelihood(self, docs: list[Doc]) -> list[ModelResponse]:
+        return self._loglikelihood(docs, rolling=False)
+
+    @cached("predictions")
+    def loglikelihood_rolling(self, docs: list[Doc], override_bs=None) -> list[ModelResponse]:
+        return self._loglikelihood(docs, rolling=True)
+
+    def _loglikelihood(self, docs: list[Doc], rolling: bool = False) -> list[ModelResponse]:
```

Comment:
Grouped logic of both functions, and separated the API function from the logic to avoid breaking changes in people's pipeline if we change the core logic